ldk/javascript/src/network/index.ts

```diff
@@ -29,6 +28,64 @@ export interface HTTPResponse {
   headers: Record<string, string[]>;
 }
 
+/**
+ * a simplified representation of a callback which take error
+ */
+export type CallbackError = (error: Error | undefined) => void;
+
+/**
+ * configuration object to configure a websocket handshake
+ */
+export interface SocketConfiguration {
+  /**
+   * websocket server endpoint url: '{schema}://{host}:{port}' - schema could be ws or wss
+   */
+  url: string;
+  /**
+   * collection of the handshake headers
+   */
+  headers?: Record<string, string[]>;
+  /**
+   * specifies if compression is used
+   */
+  useCompression?: boolean;
+  /**
+   * specifies the client's requested subprotocols
+   */
+  subprotocols?: Array<string>;
+}
+
+/**
+ * Object to communicate with the websocket
+ */
+export interface Socket {
+  /**
+   * writes message to a websocket
+   * @param message text or data message
+   * @param callback function to call if error occurred
+   */
+  writeMessage(message: string | Uint8Array, callback: CallbackError): Promise<void>;
+  /**
+   * closes websocket
+   * @param callback function to call if error occurred
+   */
+  close(callback: CallbackError): Promise<void>;
```

Comment:
Can we get the error to reject the promise instead?

Reply:
Done